src/views/ChartPage.vue

```diff
@@ -0,0 +1,463 @@
+          </div>
+          <div class="form-check">
+            <input class="form-check-input" type="checkbox" value="spent" id="spent" v-model="typeInformationBar">
+            <label class="form-check-label" for="spent">{{ $t('SPENT') }}</label>
+          </div>
+          <div class="form-check">
+            <input class="form-check-input" type="checkbox" value="available" id="available" v-model="typeInformationBar">
+            <label class="form-check-label" for="available">{{ $t('AVAILABLE') }}</label>
+          </div>
+        </div>
+        <div class="multiselect">
+          <Multiselect
+            v-on:change="recalculate"
+            v-model="masterCategoryId"
+            :groups="true"
+            :searchable="true"
+            :options="masterCategories"
+            :noResultsText="$t('NO_RESULT_FOUND')"
+            :placeholder="$t('SELECT_MASTER_CATEGORY')"
+            :show-labels="false"
+            @select="recalculate"
+          />
+        </div>
+      </div>
+
+      <div class="placeholder bottom">
+          <NavMenu :page="'chart'" />
+      </div>
+      <div class="footer fixed">
+          <NavMenu :page="'chart'" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+import { redirectToLoginPageIfNotLogged } from '@/router'
+import BarChart from '@/components/charts/BarChart.vue'
+import PieChart from '@/components/charts/PieChart.vue'
+import NavMenu from '@/components/NavigationMenu.vue'
+import StoreHandler from '@/store/StoreHandler'
+import BudgetDataService from '@/services/BudgetDataService'
+import { CategoryDataList, Budget, GroupSelectOption } from '@/model/model'
+import Utils from '@/utils/Utils'
+import Time from '@/utils/Time'
+import Multiselect from '@vueform/multiselect'
+
+interface ChartPageData {
+    chartData: {
+      labels: string[];
+      datasets: {
+        label: string;
+        backgroundColor: string[];
+        data: number[];
+      }[];
+    };
+    categoryDataList: CategoryDataList;
+    pieChartData: {
+      labels: string[];
+      datasets: {
+        backgroundColor: string[];
+        data: number[];
+      }[];
+    };
+    typeInformationPie: string;
+    typeInformationBar: string[];
+    masterCategoryId: string;
+    currentGraph: string;
+    budgetMonth: number;
+}
+
+export default defineComponent({
+  name: 'Login',
+  components: {
+    BarChart,
+    PieChart,
+    NavMenu,
+    Multiselect
+  },
+  props: { },
+  beforeCreate: async function () {
+    redirectToLoginPageIfNotLogged(this.$store)
+  },
+  created: async function () {
+    StoreHandler.initStore(this.$store)
+    await this.getBudgetData()
+    this.recalculate()
+  },
+  watch: {
+    typeInformationPie: function () {
+      this.drawPieChart()
+    },
+    typeInformationBar: function () {
+      this.drawBarChart()
+    }
+  },
+  data (): ChartPageData {
+    return {
+      chartData: {
```

Comment:
what is the difference between chartData and pieChartData

Reply:
i clarified it : chartData is now barChartData. barCahrt and pieChart need different datas